obj[] movable_objects
obj[] walls
obj[] surfaces
